openlibrary/plugins/upstream/covers.py

```diff
@@ -60,8 +60,8 @@ def upload(self, key, i):
         """Uploads a cover to coverstore and returns the response."""
         olid = key.split("/")[-1]
 
-        if i.file is not None and hasattr(i.file, 'value'):
-            data = i.file.value
+        if i.file is not None and hasattr(i.file, 'file'):
+            data = i.file.file.read()
```

Comment:
I'm not sure how big the cover uploads are in practice, so perhaps this isn't a concern, but the requests.post() method accepts a file-like object, so rather than taking this file-like object and reading its entire contents into memory, only to wrap it in BytesIO and send it back out again, it would probably be more efficient to just pass the same file-like object through.

Reply:
As usual, this is a good point well made. I have changed it to remove remove the pointless reading-and-wrapping. Thanks, @tfmorris. :)